dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -1394,18 +1406,19 @@ public String getTableToolContent(@ApiParam(hidden = true) @Parameter(hidden = t
         }
 
         // json in db cleared after a refresh
-        if (workflowVersion.getToolTableJson() != null) {
-            return workflowVersion.getToolTableJson();
+        String toolTableJson = workflowVersion.getToolTableJson();
+        if (toolTableJson != null && (toolTableJson.contains("specifier") || "[]".equals(toolTableJson))) {
```

Comment:
Can you please add a comment explaining why you're also checking for `"specifier"` and an empty array?

Isn't an empty array valid, e.g., if a workflow doesn't reference any Docker images (not recommended, but legal).

Reply:
Comment added!

Yeah, an empty array is valid. I added that check because an empty array would be missing the `"specifier"` key, but I figured it would be wasteful to retrieve the toolTableJson again since it'll just return an empty array again (because like you mentioned, the workflow doesn't reference any images).